source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -455,35 +458,63 @@ namespace AdaptiveNamespace
     }
 
     void XamlBuilder::ApplyBackgroundToRoot(_In_ ABI::Windows::UI::Xaml::Controls::IPanel* rootPanel,
-                                            _In_ HSTRING url,
+                                            _In_ IAdaptiveBackgroundImage* backgroundImage,
                                             _In_ IAdaptiveRenderContext* renderContext,
                                             _In_ IAdaptiveRenderArgs* renderArgs)
     {
         // In order to reuse the image creation code paths, we simply create an adaptive card
         // image element and then build that into xaml and apply to the root.
         ComPtr<IAdaptiveImage> adaptiveImage;
+        HSTRING url;
         THROW_IF_FAILED(MakeAndInitialize<AdaptiveImage>(&adaptiveImage));
+        backgroundImage->get_Url(&url);
         adaptiveImage->put_Url(url);
 
         ComPtr<IAdaptiveCardElement> adaptiveCardElement;
         THROW_IF_FAILED(adaptiveImage.As(&adaptiveCardElement));
-        ComPtr<IUIElement> backgroundImage;
-        BuildImage(adaptiveCardElement.Get(), renderContext, renderArgs, &backgroundImage);
-        if (backgroundImage == nullptr)
+
+        ComPtr<IUIElement> background;
+        BuildImage(adaptiveCardElement.Get(), renderContext, renderArgs, &background);
+        if (background == nullptr)
         {
             return;
         }
 
-        // All background images should be stretched to fill the whole card.
         ComPtr<IImage> xamlImage;
-        THROW_IF_FAILED(backgroundImage.As(&xamlImage));
-        THROW_IF_FAILED(xamlImage->put_Stretch(Stretch::Stretch_UniformToFill));
+        THROW_IF_FAILED(background.As(&xamlImage));
+
+        ABI::AdaptiveNamespace::BackgroundImageMode mode;
+        backgroundImage->get_Mode(&mode);
 
+        // Apply Background Image Mode
         ComPtr<IFrameworkElement> backgroundAsFrameworkElement;
-        THROW_IF_FAILED(xamlImage.As(&backgroundAsFrameworkElement));
-        THROW_IF_FAILED(backgroundAsFrameworkElement->put_VerticalAlignment(VerticalAlignment_Stretch));
+        switch (mode)
+        {
+        case ABI::AdaptiveNamespace::BackgroundImageMode::Stretch:
+            // Ignored: horizontalAlignment, verticalAlignment
+            THROW_IF_FAILED(xamlImage->put_Stretch(Stretch::Stretch_UniformToFill));
+
+            THROW_IF_FAILED(xamlImage.As(&backgroundAsFrameworkElement));
+            THROW_IF_FAILED(backgroundAsFrameworkElement->put_VerticalAlignment(VerticalAlignment_Stretch));
+            break;
+        default:
```

Comment:
>default: [](start = 8, length = 8)

Make this explicit so we can see what values this is handling:

        case ABI::AdaptiveNamespace::BackgroundImageMode::RepeatHorizontally:
        case ABI::AdaptiveNamespace::BackgroundImageMode::RepeatVertically:
        case ABI::AdaptiveNamespace::BackgroundImageMode::Repeat: